docs/readme/error_handling.md

```diff
@@ -0,0 +1,22 @@
+# Error Handling
+
+In SITUP Pipelines Errors should be handled by implementation of plugin components and should not throw any uncaught runtime exceptions. If thrown the pipeline will stop immediately or halt eventually based on the component that throws the exception.
+
+Below are the different scenario of error,
+
+## Single Pipeline
+Single pipeline is a pipeline that doesn't use the pipeline connectors.
+
+* If an exception encountered when creating a pipeline, the pipeline stops and exits using `system.exit(1)`. 
+* If an exception encountered during pipeline `start()` method, the pipeline stops with appropriate logs. 
+* If an exception encountered in Pipeline `processors` or `sinks` during runtime, the pipeline stops with appropriate logs.
+
+## Connected Pipelines
+
+Connected Pipelines is scenario where two or more pipelines connected using the Pipeline Connector.
+
+
+* If an exception encountered when creating any of the connected pipelines, all the connected pipelines stop and exits using `system.exit(1)`. 
+* If an exception encountered during pipeline `start()` method in any of the connected pipelines, the pipeline that encountered the exception will stop and other pipelines will run and will not receive or process any data as the one of the connected pipelines is unavailable.
+* If an exception encountered in Pipeline `processors` or `sinks` during runtime in any of the connected pipelines, the pipeline that encountered the exception will stop and other pipelines will run and will not receive or process any data as the one of the connected pipelines is unavailable.
```

Comment:
The other pipelines will eventually shutdown

Reply:
They won't shutdown right, they will halt because the buffer will get full?

Reply:
Yes, but eventually they will shutdown; I think we discussed this previously as well; 
Assume A -> B -> C -> ES, ES fails or processor in C fails -> this will trigger shutdown of C pipeline, which means it will stop the source which is the PipelineConnector of B & C, B will fail to write data (i.e. fail to write to its sink), triggers B.shutdown -> similarly A.shutdown

Reply:
Got it.